python-client/giskard/models/cache/cache.py

```diff
@@ -29,29 +30,45 @@ class ModelCache:
 
     vectorized_get_cache_or_na = None
 
-    def __init__(self, model_type: SupportedModelTypes, id: Optional[str] = None, cache_dir: Path = None):
+    def __init__(self, model_type: SupportedModelTypes, id: Optional[str] = None, cache_dir: Optional[Path] = None):
         self.id = id or str(uuid.uuid4())
         self.prediction_cache = dict()
-        self.cache_dir = cache_dir or Path(settings.home_dir / settings.cache_dir / "global/prediction_cache" / self.id)
-
-        if id is not None:
-            if (self.cache_dir / CACHE_CSV_FILENAME).exists():
-                with open(self.cache_dir / CACHE_CSV_FILENAME, "r") as pred_f:
-                    reader = csv.reader(pred_f)
-                    for row in reader:
-                        if model_type == SupportedModelTypes.TEXT_GENERATION:
-                            self.prediction_cache[row[0]] = row[1:]
-                        elif model_type == SupportedModelTypes.REGRESSION:
-                            self.prediction_cache[row[0]] = float(row[1])
-                        else:
-                            self.prediction_cache[row[0]] = [float(i) for i in row[1:]]
-
+        self.cache_dir = cache_dir or Path(
+            settings.home_dir / settings.cache_dir / "global" / "prediction_cache" / self.id
+        )
+        self.cache_file = self.cache_dir / CACHE_CSV_FILENAME
         self.vectorized_get_cache_or_na = np.vectorize(self.get_cache_or_na, otypes=[object])
+        self.model_type = model_type
+        self._warmed_up = False
+
+    def warm_up_from_disk(self):
```

Comment:
Why not warm up at creation time instead of `read_from_cache` ?

Reply:
Because I don’t want to warm up if the cache is disabled. But it’s still better to create the `ModelCache` object in the model, so that it can be used if the cache is enabled at a later time.

Reply:
I think in case cache is disabled the cleanest will be not to initialize `_cache` property at all and then warm it up in the ModelCache constructor depending on a cache type (in memory only vs FS backed), WDYT?

Reply:
But cache can be disabled temporarily. So if we don’t initialize the `_cache` property at all, we would need to check if it exists at every prediction, initializing it there and possibly warm it up at prediction time, in addition to checking that cache is enabled.

In any case I would avoid doing expensive and persistent operations upon construction of the instance.

Reply:
ok let's keep it this way